File: 1422 maximum-score-after-splitting-a-string/solution.java
class Solution {
    public int maxScore(String s) {
        int ones=0,zeros=0,max=0;
        for(char c:s.toCharArray()) if(c=='1') ones++;

        for(int i=0;i<s.length()-1;i++) {
            if(s.charAt(i)=='0') zeros++;
            else ones--;
            max=Math.max(max,ones+zeros);
        }
        return max;
    }
}

File: 1422 maximum-score-after-splitting-a-string/solution.cs
public class Solution {
    public int MaxScore(string s) {
        int o=0,z=0,max=0;
        foreach(char i in s) if(i=='1') o++;
        for(int i=0;i<s.Length-1;i++) {
            if(s[i]=='0') z++;
            else o--;
            max=Math.Max(max,o+z);
        } 
        return max;
    }
}
